src/pagination/mod.rs

```diff
@@ -136,23 +135,45 @@ fn interpret_permalink(
     page_num: usize,
     index: Option<&liquid::value::Value>,
 ) -> Result<String> {
-    Ok(if page_num == 1 {
-        if let Some(index) = index {
-            let include: &str = config.include.into();
-            format!("{}/{}", include, index_to_string(&index))
-        } else {
-            doc.url_path.clone()
-        }
+    let mut attributes = document::permalink_attributes(&doc.front, &doc.file_path);
+    let permalink = permalink::explode_permalink(&config.front_permalink, &attributes)?;
+    let permalink_path = std::path::Path::new(&permalink);
+    let mut pagination_root = permalink_path.extension().map_or_else(
+        || permalink.clone(),
+        |os_str| {
+            permalink
+                .trim_end_matches(&format!(".{}", os_str.to_string_lossy()))
+                .to_string()
+        },
+    );
+    // avoid "//"
```

Comment:
`//all/_p/2/` is not working, it gives us a link `all/_p/2` which leads to `http://all/_p/2/`.

Whereas `/all/_p/2/` gives us `http://localhost:3030/all/_p/2/`

Reply:
We do not support user-configured relative paths out of simplicity *but* the post-processed permalinks are made relative in case the generated site is not in the user's root.

I recommend doing an explicit `/{{ page.permalink}}` or even better `{{ site.base_url }}/{{ page.permalink }}`.

Reply:
Ok I see! Make sense. Removing that!